### ISSUE TITLE
Update the UI with notes on how to configure kubectl to work with rancher/ssl/k8s

### ISSUE THREAD

**Issue by sangeethah** [rancher/rancher#7235]
Rancher Version:v1.3.0-rc2 using self signed certs.

rancher-server is started with the following command - sudo docker run -d --restart=unless-stopped --name=rancher-server -v :/var/lib/rancher/etc/ssl/ca.crt rancher/server:v1.3.0-rc2

It is hosted behind nginx server that acts as a reverse proxy.

On the host , certs have to be copied to /var/lib/rancher/etc/ssl/ca.crt before running the host registration url which will already include -v /var/lib/rancher:/var/lib/rancher

Able to add hosts successfully to "k8s" environment.
All infrastructure stacks comes up successfully.
Able to create services using Kubectl from UI.

Not able to connect to tis rancher-server setup using the generated config.
we get the following error:
```Unable to connect to the server: x509: certificate signed by unknown authority```

**Comment by wlan0** [comment]
Setting the configuration in kubeconfig is not fixing the issue. I need to read up and understand if any kubernetes api server changes are required.

**Comment by wlan0** [comment]
The steps to get this working are  - 

1. Download the ca.crt, ca-cert.crt, and ca-key.pem to the local machine that is running kubectl
2. Set the ca-cert and ca-key to your current config. For eg, If the config name is "Default", then run

```
kubectl config set-credentials Default --client-certificate=~/.kube/admin.crt --embed-certs=true
kubectl config set-credentials Default --client-key=~/.kube/admin.key
```

You can now execute kubectl commands from your remote kubectl client using the following command

```
kubectl --certificate-authority=/path/to/ca $CMD

# for eg. this command will return both client and server version
kubectl --certificate-authority=/path/to/ca version
```

**Comment by sangeethah** [comment]
In my case , environment was "hello" , I used the following commands to set the cert and key(Note I had to use the full path to the cert and key):

```
 ./kubectl config set-credentials hello --client-certificate=/Users/sangeethahariharan1/.kube/admin.crt --embed-certs=true
./kubectl config set-credentials hello --client-key=/Users/sangeethahariharan1/.kube/admin.key
user "hello" set.
```
Able to connect using the following command:

```
$ ./kubectl --certificate-authority=/Users/sangeethahariharan1/.kube/admin.crt version
Client Version: version.Info{Major:"1", Minor:"4", GitVersion:"v1.4.0", GitCommit:"a16c0a7f71a6f93c7e0f222d961f4675cd97a46b", GitTreeState:"clean", BuildDate:"2016-09-26T18:16:57Z", GoVersion:"go1.6.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"5+", GitVersion:"v1.5.1-rancher1", GitCommit:"24adf52d6f18be9071dc3194b4dca4a0d4e35bb9", GitTreeState:"clean", BuildDate:"2016-12-15T05:28:48Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"linux/amd64"

```

**Comment by sangeethah** [comment]
Cert needs to be set for the cluster using the following command before executing any kubectl commands:
```./kubectl config set-cluster <env name> --certificate-authority=<path-to-cert>```

Other steps mentioned in the previous comments are not needed.

**Comment by wlan0** [comment]
These steps need to be documented in the docs/the UI should be updated to provide the self signed ca as a part of the kubeconfig.

**Comment by deniseschannon** [comment]
API needs to return what the cert is in order for the UI to include this in the kube config.

**Comment by LLParse** [comment]
I ran into a similar issue when trying to integrate OpenShift with our Kubernetes stack. Cert path validation never worked because we don't configure kubelets with valid certs signed by the Rancher-generated CA. Kubelet actually generates its own self-signed certs in this case (there is no longer an option to expose it over plaintext).

Is it also possible to runtime reconfigure the kubelet(s) self-signed certificates securing its own API (without modifying our k8s stack)?

**Comment by alena1108** [comment]
@wlan0 we can expose it via setting (it should it listed among settings.public [here](https://github.com/rancher/cattle/blob/f8010fe3a9fa46050cb0d9e2ab625a0aae561222/code/packaging/app-config/src/main/resources/META-INF/cattle/api-server/defaults.properties#L66)

The setting has to be populated from /var/lib/rancher/etc/ssl/ca.crt - the argument passed on rancher server start. This is the only tricky part.

**Comment by vincent99** [comment]
@alena1108 @wlan0 I need the setting with the cert, or what you want to say about how to get it, and what they're supposed to do with it once they have it.  All I can do with this right now is show

![image](https://cloud.githubusercontent.com/assets/753917/22382614/a5ac1d36-e483-11e6-8c12-762201ca95a1.png)

**Comment by wlan0** [comment]
The user need to fetch the certificate authority for her self signed cert (ca.pem), and then save it in a local file called ca.pem

Before executing any kubectl commands, the user needs to set the cert in the current kube context using this command -
```
./kubectl config set-cluster <env name> --certificate-authority=<path-to-cert>
```

Ensure that the path to cert is the complete path starting from root

**Comment by laszlocph** [comment]
I was able to get the root certificate file, from one of the k8s containers (/etc/kubernetes/ssl/ca.crt), and put it in the `.kube/config` file under the `certificate-authority` key but when I access the API via kubectl I get the infamous: `Unable to connect to the server: x509: certificate is valid for X, not Y`

In my case X is my internal IP, and I access it via Y, which is the public domain name. Trying internally with the same kube config, it works perfectly.

I messed around a bit with various certs, but couldn't really make the API return the cert where my public domain was added as a SAN, nor CATTLE_AGENT_IP had any control over the cert's domain.

Is there a way where I can get Rancher or the Kube API to return my custom certification?
